### DISCUSSION TITLE
Endpoint suitable for QGIS-based raster stats

### DISCUSSION THREAD

**Discussion by dchirst** [developmentseed/titiler#773] [Q&A] [answered]
Hey Vincent!

I've been using Titiler recently to stream some of our STACs to QGIS using the XYZ endpoint. Ideally, we'd like to stream the raw data so that we can perform raster statistics on the streamed COG. Trying this, I'm causing QGIS to clog up and die 😅 so I have a feeling that this isn't the right way to do this.

I know we could use the statistics endpoint, but I'd ideally like to use the QGIS functions for simplicity. 

My question is: is there a performant way to stream the raw COGs via Titiler, or is the XYZ endpoint not designed for this purpose so I should use some other solution? Would the WMS extension likely improve results?

**Comment by vincentsarago** [comment]
I guess you cannot access the COG directly?

**Reply by dchirst** [reply]
It is a possibility, yes, but we'd rather use an endpoint to reduce the number of permissions that the users will need. We might use the STAC QGIS plugin, though

**Reply by vincentsarago** [reply]
I wonder if something like https://github.com/rouault/cogserver could be better

**Reply by dchirst** [reply]
Thanks for sending that? If I may, what is the difference in functionality between cogserver and titiler? Is it that Cogserver is providing the raw tiff, whereas titiler is turning it into pngs on the fly for the XYZ endpoint?

**Reply by vincentsarago** [reply, marked as answer]
I think the cogserver transform anyfile in COG (virtually) so basically you could create an API using this. The XYZ endpoint involve reprojection so it might not be the most efficient method. 

as you mentioned you could also use the WMS or the WMTS endpoint but it might be similar.

**Reply by dchirst** [reply]
Thanks so much for your time Vincent!
